main.rb

```diff
@@ -0,0 +1,151 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    array = is_a?(Range) ? to_a : self
+    array.length.times { |index| yield(array[index]) }
+    array
+  end
+
+  def my_each_with_index
+    return enum_for unless block_given?
+
+    array = is_a?(Range) ? to_a : self
+    array.length.times { |index| yield(array[index], index) }
+    array
+  end
+
+  def my_select
+    return enum_for unless block_given?
+
+    array = []
+    my_each { |index| array.push(index) if yield(index) }
+    array
+  end
+
+  def my_all?(*arguments)
+    if !arguments[0].nil?
+      my_each { |index| return true unless arguments[0] == index }
+    elsif !block_given?
+      my_each { |index| return true unless index }
+    else
+      my_each { |index| return true unless yield(index) }
+    end
+    false
+  end
+
+  def my_any?(*arguments)
+    if !arguments[0].nil?
+      my_each { |index| return false unless arguments[0] == index }
+    elsif !block_given?
+      my_each { |index| return false unless index }
+    else
+      my_each { |index| return false unless yield(index) }
+    end
+    true
+  end
+
+  def my_none?(*args)
+    !my_any?(*args)
+  end
+
+  def my_count(arg = nil)
+    counter = 0
+    if arg
+      my_each { |element| counter += 1 if element == arg }
+    elsif !block_given?
+      counter = length
+    elsif !arg
+      my_each { |element| counter += 1 if yield element }
+    end
+    counter
+  end
```

Comment:
#my_none? 
1. [ ] Returns true if the block never returns true for all elements
### Test 👉 `expect(array.my_none?(&true_block)).to eq(array.none?(&true_block))`
|Expected|Got|
|:---:|:---:|
|`true`|`false`|
##

2. [ ] When no block or argument is given returns false only if one of the collection members is true
### Test 👉 `expect(true_array.my_none?).to be false`
|Expected|Got|
|:---:|:---:|
|`false`|`true`|
##
3. [ ] When a Regex is passed as an argument returns false if any of the collection matches the Regex
### Test 👉 `expect(words.my_none?(/d/)).to be words.none?(/d/)`
|Expected|Got|
|:---:|:---:|
|`false`|`true`|
##

4. [ ] When a pattern other than Regex or a Class is given returns false only if one of the collection matches the pattern
### Test 👉 `expect(words.my_none?(5)).to be words.none?(5)`
|Expected|Got|
|:---:|:---:|
|`false`|`true`|
##